### PR TITLE
ConfigurationResolver - Handle empty "rules" value

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3019

### CODE REVIEW

File: tests/Console/Command/FixCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+use PhpCsFixer\Console\Application;
+use PhpCsFixer\Console\Command\FixCommand;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\Command\FixCommand
```
Comment (SpacePossum) [inline tests/Console/Command/FixCommandTest.php]: not sure about this

File: tests/Console/Command/FixCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        $this->application = new Application();
+    }
+
+    public function testEmptyRulesValue()
+    {
+        $this->doTestExecute(
+            array('--rules' => ''),
+            array(
+                'class' => InvalidConfigurationException::class,
+                'regex' => '#^Empty rules value is not allowed\.$#',
```
Comment (keradus) [inline tests/Console/Command/FixCommandTest.php]: not sure about having same utests for 2 layers - currently, it's for command and resolver. but the logic is the same, and command is passing values to resolver.

Comment (SpacePossum) [reply tests/Console/Command/FixCommandTest.php]: Thought about it, but I went for this route.
When passing "php-cs-fixer --rule=&nbsp;&nbsp;&nbsp;" than the value of `--rule=` will be set to "" (and not "&nbsp;&nbsp;&nbsp;").
This behavior seems to be bound to SF/console and I thought it best to cover it in a test as well.
So in theory the application will never passes "&nbsp;&nbsp;&nbsp;" as value for `rules` to the configuration resolver, however people may use the configuration outside of the application so may pass it.

**Comment by keradus** [comment]
Travis needs fixing
I fixed one of the warnings, leving 2nd for you @SpacePossum

**Comment by SpacePossum** [comment]
thanks @keradus 
test update and rebased on 2.2

File: tests/Console/Command/FixCommandTest.php [hunk 3]
```diff
@@ -52,7 +52,7 @@ public function testEmptyRulesValue()
      */
     public function testEmptyFormatValue()
     {
-        $cmdTester = $this->doTestExecute(array('--using-cache' => 'not today'));
+        $cmdTester = $this->doTestExecute(array('--using-cache' => 'not today', '--rules=switch_case_semicolon_to_colon'));
```
Comment (keradus) [inline tests/Console/Command/FixCommandTest.php]: btw, why passing `--rules` fixing the issue?

Comment (SpacePossum) [reply tests/Console/Command/FixCommandTest.php]: the test check the result of fixing this file, which might have diff. with the default set on diff. PHP versions

Comment (keradus) [reply tests/Console/Command/FixCommandTest.php]: well, that change didn't fix travis.

**Comment by keradus** [comment]
not sure what is the issue, works locally:
```
~/github/PHP-CS-Fixer$ docker run -it --rm --name my-running-script -v "$PWD":/usr/src/myapp -w /usr/src/myapp php:7.2.0RC1-cli php vendor/phpunit/phpunit/phpunit tests/Console/Command/FixCommandTest.php
PHPUnit 5.7.21 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.0RC1
Configuration: /usr/src/myapp/phpunit.xml.dist

Testing PhpCsFixer\Tests\Console\Command\FixCommandTest
..                                                                                                                              2 / 2 (100%)

Time: 109 ms, Memory: 10.00MB

OK (2 tests, 4 assertions)
```

**Comment by SpacePossum** [comment]
fixed now, commit history is a bit of a mess, I'll clean up later

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**